http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java

```diff
@@ -29,161 +29,167 @@
+     * default is shared among all threads ({@link #mapper}). It is injectable here because of best
+     * practices: for further configuration support and unit testing.
+     */
+    public LightblueHttpClient(LightblueClientConfiguration configuration, ObjectMapper mapper) {
+        Objects.requireNonNull(configuration, "configuration");
+        Objects.requireNonNull(mapper, "mapper");
+
+        // Make a defensive copy because configuration is mutable. This prevents alterations to the
+        // config object from affecting this client after instantiation.
+        this.configuration = new LightblueClientConfiguration(configuration);
+        this.mapper = mapper;
+    }
+
+    /**
+     * @deprecated
+     * Use LightblueHttpClient(String configFilePath) if you want to specify a config file location not on the classpath
+     * Use LightblueHttpClient(LightblueClientConfiguration configuration) if you don't want to use config files at all
+     */
+    @Deprecated
+    public LightblueHttpClient(String dataServiceURI, String metadataServiceURI, Boolean useCertAuth) {
+        LightblueClientConfiguration configuration = new LightblueClientConfiguration();
+        configuration.setDataServiceURI(dataServiceURI);
+        configuration.setMetadataServiceURI(metadataServiceURI);
+        configuration.setUseCertAuth(useCertAuth);
+
+        this.configuration = configuration;
+        this.mapper = DEFAULT_MAPPER;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * com.redhat.lightblue.client.LightblueClient#metadata(com.redhat.lightblue
+     * .client.request.LightblueRequest)
+     */
+    @Override
+    public LightblueResponse metadata(LightblueRequest lightblueRequest) {
+        LOGGER.debug("Calling metadata service with lightblueRequest: " + lightblueRequest.toString());
+        return callService(new LightblueHttpMetadataRequest(lightblueRequest)
+        .getRestRequest(configuration.getMetadataServiceURI()));
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * com.redhat.lightblue.client.LightblueClient#data(com.redhat.lightblue.client
+     * .request.LightblueRequest)
+     */
+    @Override
+    public LightblueResponse data(LightblueRequest lightblueRequest) {
+        LOGGER.debug("Calling data service with lightblueRequest: " + lightblueRequest.toString());
+        return callService(new LightblueHttpDataRequest(lightblueRequest)
+        .getRestRequest(configuration.getDataServiceURI()));
+    }
+
+    @Override
+    public <T> T data(LightblueRequest lightblueRequest, Class<T> type) throws IOException {
+        try{
+            LightblueResponse response = data(lightblueRequest);
+
+            JsonNode objectNode = response.getJson().path("processed");
+
+            try {
+                return mapper.readValue(objectNode.traverse(), type);
+            } catch (JsonMappingException e) {
```

Comment:
Yes, the inner exception would be wrapped in the outer. I did this because in the case of JsonMappingException, both pieces of information are useful for debugging purposes. However, if the a RuntimeException were to happen from anywhere else in the "data" method, only the outer exception is relevant. 

If you are asking why I didn't just catch JsonMappingException in the outer block, it is primarily because I was trying to prevent duplicate logic. In that I would need to print "Error sending lightblue request:" in both catch blocks.